src/main/java/org/killbill/billing/plugin/api/notification/PluginTenantConfigurableConfigurationHandler.java

```diff
@@ -60,8 +60,13 @@ protected void configure(@Nullable final UUID kbTenantId) {
     public C getConfigurable(@Nullable final UUID kbTenantId) {
         // Initial configuration
         if (kbTenantId != null && !configuredTenants.contains(kbTenantId)) {
-            configure(kbTenantId);
-            configuredTenants.add(kbTenantId);
+            // Make sure to initialize the value for the tenant once
+            synchronized (configuredTenants) {
```

Comment:
It seems like adding the volatile keyword prevents the compiler from changing the order of reading/writing to the field, so it [might actually work](http://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html)